### PR TITLE
feat: gather town space as reward

### PR DESCRIPTION
**Pull request by FBalint**
TODO:

- [x] create and edit form button disable state control
- [x] api key field should be optional on edit (also depends on ongoing backend discussions)
- [x] refactoring

### CODE REVIEW

File: src/platforms/Gather/GatherConnectionStatusAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          </>
+        ),
+      }
+    } else {
+      return {
+        status: "info",
+        children: (
+          <>
+            <AlertIcon mt={0} />
+            <p>Enter API key and space URL below to set up connection.</p>
```
Comment (TomiOhl) [inline src/platforms/Gather/GatherConnectionStatusAlert.tsx:67]: Nit
```suggestion
            <p>Enter your API key and space URL below to set up a connection.</p>
```

File: src/platforms/Gather/GatherForm.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { Divider, Text } from "@chakra-ui/react"
+import GatherConnectionForm from "./GatherConnectionForm"
+import GatherGuestForm from "./GatherGuestForm"
+
+const GatherForm = () => {
+  return (
+    <>
+      <Text colorScheme="gray" fontWeight="semibold" mb="8">
+        The email addresses of eligible users will be added to the guest list for the
+        Gather space, which you can specify below.
```
Comment (TomiOhl) [inline src/platforms/Gather/GatherForm.tsx:10]:
```suggestion
        The email addresses of eligible users will be added to the guest list of the
        Gather space, which you can specify below.
```

File: src/platforms/Gather/GatherConnectionForm.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      <GatherConnectionStatusAlert
+        isLoading={accessLoading}
+        success={accessSuccess}
+        error={accessError}
+      />
+
+      <FormControl pt={{ md: 0.5 }} isInvalid={!!errors?.gatherApiKey}>
+        <Stack>
+          <HStack>
+            <Text as="span">API key:</Text>
```
Comment (dovalid) [inline src/platforms/Gather/GatherConnectionForm.tsx:52]: Should use FormLabels, so the styling is consistent and the input is focusable by clicking on the label too

File: src/platforms/Gather/GatherConnectionForm.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+          <Link
+            href="https://gather.town/apiKeys"
+            textDecoration="underline"
+            isExternal
+          >
+            here
+          </Link>
+          .
+        </Text>
+        <FormErrorMessage>{errors?.gatherApiKey?.message}</FormErrorMessage>
```
Comment (dovalid) [inline src/platforms/Gather/GatherConnectionForm.tsx:85]: Should use the common FormErrorMessage component, so it animates

File: src/platforms/Gather/GatherConnectionForm.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+              {...register("gatherApiKey", {
+                required: "This field is required",
+              })}
+            />
+            <InputRightElement onClick={() => setShowApiKey(!showApiKey)}>
+              {showApiKey ? <EyeClosed /> : <Eye />}
+            </InputRightElement>
+          </InputGroup>
+        </Stack>
+        <Text as="small" colorScheme="gray">
```
Comment (dovalid) [inline src/platforms/Gather/GatherConnectionForm.tsx:74]: Should use FormHelperText

File: src/platforms/Gather/GatherConnectionForm.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+          <InputGroup>
+            <Input
+              autoComplete="off"
+              type={showApiKey ? "text" : "password"}
+              {...register("gatherApiKey", {
+                required: "This field is required",
+              })}
+            />
+            <InputRightElement onClick={() => setShowApiKey(!showApiKey)}>
+              {showApiKey ? <EyeClosed /> : <Eye />}
```
Comment (dovalid) [inline src/platforms/Gather/GatherConnectionForm.tsx:70]: Would be great to wrap the icons in an IconButton so we have hover/focus states and they're focusable. Or at least set a cursor: pointer style

File: src/platforms/Gather/GatherConnectionForm.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+              placement="top"
+              hasArrow
+            >
+              <Icon as={Question} color="GrayText" />
+            </Tooltip>
+          </HStack>
+          <InputGroup>
+            <Input
+              autoComplete="off"
+              type={showApiKey ? "text" : "password"}
```
Comment (dovalid) [inline src/platforms/Gather/GatherConnectionForm.tsx:64]: When being a pw and clicking next, the browser asked me if I'd want to save it, idk if we can prevent that

File: src/platforms/components/GatherPreview.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+import { useWatch } from "react-hook-form"
+import PlatformPreview from "./PlatformPreview"
+
+const GatherPreview = () => {
+  const platformGuildData = useWatch({
+    name: "rolePlatforms.0.guildPlatform.platformGuildData",
+  })
+
+  return (
+    <PlatformPreview
```
Comment (dovalid) [inline src/platforms/components/GatherPreview.tsx:11]: AvailabilitySetup is not shown for Gather, while it should be. We have to add it elsewhere, just commenting here

File: src/platforms/Gather/GatherConnectionForm.tsx (hunk 7 above)
Comment (FBalint) [reply src/platforms/Gather/GatherConnectionForm.tsx:67]: The `autoComplete='off'` attribute should theoretically address this issue, yet it appears to be ineffective. Should we then default to a standard text input?

Most solutions to this often seem excessively complicated. We could apply the `-webkit-text-security: disc` property in CSS, which appears to be widely supported on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-text-security#browser_compatibility) and [Can I Use](https://caniuse.com/?search=-webkit-text-security), though it's still considered non-standard.

File: src/platforms/Gather/GatherGuestForm.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+import { SectionTitle } from "components/common/Section"
+import { Question } from "phosphor-react"
+import { useFormContext } from "react-hook-form"
+
+const GatherGuestForm = () => {
+  const { register } = useFormContext<AddGatherFormType>()
+
+  return (
+    <>
+      <FormControl>
```
Comment (dovalid) [inline src/platforms/Gather/GatherGuestForm.tsx:20]: We shouldn't use FormControl as a high level wrapper like this and have multiple inputs in one. We should replace the Stacks around the labels + inputs below instead to be individual FormControls

**Pull request merged**